docs/DeveloperGuide.md

```diff
@@ -154,6 +154,51 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Show assignment list feature
+
+#### Implementation
+
+The show assignment mechanism is facilitated by `AddressBook`, where the specified person's assignment list is stored internally under `assignments` This `assignments` is retrieved or updated by the following methods:
+* `AddressBook#getAssignmentList()`
+* `AddressBook#updateAssignmentList(Person person)`  —  where `person` is the specified person.
+
+These methods are exposed in the `Model` interface as `Model#getFilteredAssignmentList()` and `Model#updateFilteredAssignmentList(Person person)` respectively.
+
+Given below is an example usage scenario and how the show assignment mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `assignments` will be initialized with an empty list.
+
+![AssignmentsState0](images/AssignmentsState0.png)
+
+Step 2. The user inputs `a-show 2` command to display the 2nd person's assignment list in the address book. The `show` command will then call `Model#updateFilterdAssignmentList(person)`, whereby `person` varaible is the 2nd person n the address book.
+This causes the `assignments` in `AddressBook` to be replaced with the 2nd person's assignment list.
+
+![AssignmentsState1](images/AssignmentsState1.png)
+
+
+Step 3. When `assignments` is updated, it is retrieved by the `Logic` using `Model#getFilteredAssignmentList()` to input into the assignment panel of the `Ui`
+This results in the assignment list panel to display the assignments of the person.
+
+![AssignmentsState2](images/AssignmentsState2.png)
+
+
+Step 4. The user decides to modify the assignment list of the person by using either `a-add`, `a-done` or `a-delete` command. This will result in the assignment list in the person to be modified.
+The command will the call `Model#updateFilteredAssignmentList(person)` to get the recent updated assignment list to replace `assignments`.
+
+![AssignmentsState3](images/AssignmentsState3.png)
+
+
+Step 5. Step 3 is repeated to show the recent updated assignment list.
+
+![AssignmentsState4](images/AssignmentsState4.png)
+
+
+####Design considerations:
+The assignment list of the specified person is stored in `AddressBook` rather than `ModelManger`
```

Comment:
Actually, ModelManager has the same data as addressbook. The fields in `ModelManager` points to the same list in addressbook. any change in addressbook person/assignment list will automatically be reflected in ModelManager.

Instead of this, perhaps you can analyse the design considerations of the use of `activePerson` field in addressbook to  point out whose assignments are currently displayed when you do `updateAssignmentList`

activePerson is mainly used for deletePerson command. If the person  you delete doesn't have his assignments displayed, then it proceeds normally. What if the person whose assignments are currently displayed is deleted?